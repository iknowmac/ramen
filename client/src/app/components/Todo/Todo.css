.todo {
  font-family: proxima-nova,"Helvetica Neue",Helvetica,Roboto,Arial,sans-serif;
  max-width: 300px;
  margin: 20px auto 20px auto;
}

.todo__header {
    color: #484848;
}

.todo__list {
  padding:0;
  list-style-type: none;
}

.todo__item {
  background: #484848;
  padding: 10px;
  margin-bottom: 5px;
  border-radius:3px;
  color: white;
  &:hover {
    cursor: pointer;
  }
}

.todo__item--done{
    display: inline-block;
    margin-left: 10px;
    margin-bottom: 1px;
  &:after {
    content: '';
    display: block;
    width: 5px;
    height: 10px;
    border: solid white;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  }
}

.todo__input {
  height:5px;
  min-width: 200px;
  margin-right: 10px;
  padding:10px;
  font-size:12px;
}

.todo__button {
	box-shadow:inset 0px 1px 0px 0px #ffffff;
	background:linear-gradient(to top, #f9f9f9 5%, #e9e9e9 100%);
	background-color:#f9f9f9;
	border-radius:3px;
	border:1px solid #dcdcdc;
	display:inline-block;
	cursor:pointer;
	color:#666666;
	font-family:Arial;
	font-size:12px;
	font-weight:bold;
	padding:8px 20px;
	text-decoration:none;
	text-shadow:0px 1px 0px #ffffff;
}