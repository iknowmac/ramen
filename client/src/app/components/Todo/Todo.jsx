'use strict';
import React, { PropTypes } from 'react';

const Todo = (props) => (
  <li className="todo__item" onClick={props.onClick}>
    {props.text} {props.completed ? <span className="todo__item--done"></span> : false}
  </li>
);

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired
};

export default Todo;
