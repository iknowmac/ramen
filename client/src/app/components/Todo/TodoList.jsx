'use strict';
import React, { PropTypes } from 'react';
import Todo from './Todo';
import './Todo.css';

const TodoList = (props) => (
  <ul className="todo__list">
    {props.todos.map(todo =>
      <Todo key={todo.id} {...todo} onClick={() => props.onClick(todo.id) } />
    ) }
  </ul>
);

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired).isRequired,
  onClick: PropTypes.func.isRequired
};

export default TodoList;
