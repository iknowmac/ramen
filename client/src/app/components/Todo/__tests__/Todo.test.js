'use strict';
import React from 'react';
import Todo from '../Todo';
import renderer from 'react-test-renderer';

describe('Todo', () => {

    const mockEvent = new Event("onClick");
    mockEvent.preventDefault = (e) => { }

    const mockHandler = () => { }

    it('renders correctly', () => {
        const tree = renderer.create(
            <Todo text="This is a todo item" completed={false} onClick={mockHandler} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with span when completed is true', () => {
        const tree = renderer.create(
            <Todo text="This is a todo item" completed={true} onClick={mockHandler} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('fires event for supplied onClick handler', () => {
        const component = renderer.create(
            <Todo text="This is a todo item" completed={false} onClick={mockHandler} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // manually trigger the callback
        tree.props.onClick(mockEvent);
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

});