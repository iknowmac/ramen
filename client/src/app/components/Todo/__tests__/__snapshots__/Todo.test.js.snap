exports[`Todo fires event for supplied onClick handler 1`] = `
<li
  className="todo__item"
  onClick={[Function]}>
  This is a todo item
   
</li>
`;

exports[`Todo fires event for supplied onClick handler 2`] = `
<li
  className="todo__item"
  onClick={[Function]}>
  This is a todo item
   
</li>
`;

exports[`Todo renders correctly 1`] = `
<li
  className="todo__item"
  onClick={[Function]}>
  This is a todo item
   
</li>
`;

exports[`Todo renders with span when completed is true 1`] = `
<li
  className="todo__item"
  onClick={[Function]}>
  This is a todo item
   
  <span
    className="todo__item--done" />
</li>
`;
